ruby-challenges.rb

```diff
@@ -19,16 +30,65 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def return_sum array
+    # create a method called return_sum that has an array as a parameter
+    array.inject { |sum, num| sum + num}
+    # use the enumerator method .inject to take the zero index of the array and the first index and adds them together, continuing down the array until all numbers have passed through the block
+    # returns the sum of all the numbers in the array 
+end 
 
+return_sum nums_array1
+return_sum nums_array2
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+class Bike
+    attr_accessor :model, :wheels, :current_speed
+    def initialize(model, wheels, current_speed)
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+    end
+    def bike_info
+        p "The Trek bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end
+end
 
+my_bike = Bike.new("schwinn",2,10)
+my_bike.bike_info
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster(10) => 10
 # Expected output example: my_bike.pedal_faster(10) => 20
 # Expected output example: my_bike.brake(25) => 0
+
+class Bike
+    # attr_accessor :model, :wheels, :current_speed, 
+    def initialize(model, wheels, current_speed)
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+        @pedal_faster = 10
+        @brake = 0
+    end
+    def bike_info
+        p "The Trek bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end
+    def pedal_faster(pedal_faster)
+        p @current_speed = @current_speed + @pedal_faster
+    end
+    def pedal_brake(brake)
+        if @current_speed >= 0
```

Comment:
There's a conditional statement addition here that would make this class a bit more robust. 

```ruby
def pedal_brake(brake)
        if @current_speed - brake > 0
           p @current_speed = @current_speed - @brake
        else 
           @current_speed = 0
end
```